src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -12,9 +12,23 @@ namespace Stratis.Bitcoin.Consensus
     /// <summary>
     /// TODO comment this interface
     /// </summary>
-    public interface IChainedHeaderValidator
+    public interface IChainedHeaderTreeValidator
```

Comment:
Why renamed? New name suggests that tree is being validated

Reply:
Hmm good point, it was rename to suggest this is a validator that is used by the tree, however this validator might also validate blocks, but I get the confusion I'll revert.